kb/development/mbedtls-coding-standards.md

```diff
@@ -318,6 +318,35 @@ Only use macros if they improve readability or maintainability, preferably both.
 
 If possible, use the C core language rather than macros. For example, if an expression is used often, a `static inline` function is better because it provides type checks, allows the compiler to keep the function call when optimizing for size, and avoids problems with arguments evaluated more than once. However, if the expression needs to be a compile-time constant when its parameters are, this is a good reason to use a macro.
 
+### Macro definition hygiene
+
+When the code contains a macro call `MBEDTLS_FOO( x, y )`, it should behave as much as possible as if `MBEDTLS_FOO` was a function. Deviate from this only to the extent necessary to make the macro practical.
+
+If the arguments of a macro are C expressions (they usually are), put parentheses around the argument in the expansion. For example:
+```c
+#define FOO_SIZE( bits ) ( ( (bits) + 7 ) / 8 + 4)
+```
+The expansion contains `( (bits) + 7 )`, not `bits + 7`, so that a call like `FOO_SIZE( x << 3 )` is parsed correctly. As an exception, it's ok to omit parentheses if the argument is directly passed to a function argument (or comma operator): `#define A( x ) f( x, 0 )` is acceptable.
+
+If the expansion of a macro is a C expression, put parentheses around the expansion. Continuing the example above, this is so that a call like `FOO_SIZE( x ) * 2` is parsed correctly. As an exception, it's ok to omit parentheses if the expansion is a function call or other lowest-precedence operator: `#define A( x ) f( x, 0 )` is acceptable.
```

Comment:
Do you really mean lowest here?

Reply:
Yes, right, the other low. (I think that was copypasta from the previous paragraph and then I rewrote the previous paragraph.)